receiver/zookeeperreceiver/testdata/scraper/correctness-v3.4.14.json

```diff
@@ -0,0 +1,219 @@
+{
```

Comment:
How is this file generated? Do we have any helper tools for this?

Did you consider yaml just for better readability? maybe with strict type restrictions if that's the concern...

Reply:
> How is this file generated? Do we have any helper tools for this?

There are complementary functions for reading and writing these files in the scrapertest package. Typical dev pattern for adding a new test would be:

1. Write test as you would otherwise, to the point where you have "actual" result:
```go
actualMetrics, err := scraper.scrape(ctx)
```

2. Use scrapertest to write a new file:
```go
actual, err := scraper.scrape(ctx)
expectedFile := filepath.Join("testdata", "scraper", "new-test.json")

scrapertest.WriteExpected(expectedFile, actual) // TODO remove this line
```

3. Review the new file to ensure it is expected result
4. Replace `WriteExpected` with `ReadExpected`, and add comparison
```go
actual, err := scraper.scrape(ctx)
expectedFile := filepath.Join("testdata", "scraper", "new-test.json")

// scrapertest.WriteExpected(expectedFile, actual) // TODO remove this line

expected, err := scrapertest.ReadExpected(expectedFile)
require.NoError(t, err)

eMetricSlice := expected.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics()
aMetricSlice := actual.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics()

require.NoError(t, scrapertest.CompareMetricSlices(eMetricSlice, aMetricSlice))
```

---

Using your suggestion of having `scrapertest.CompareMetrics(expected, actual pdata.Metrics) error:
```go
actual, err := scraper.scrape(ctx)
expectedFile := filepath.Join("testdata", "scraper", "new-test.json")
// scrapertest.WriteExpected(expectedFile, actual) // TODO remove this line
expected, err := scrapertest.ReadExpected(expectedFile)
require.NoError(t, err)
require.NoError(t, scrapertest.CompareMetrics(expected, actual))
```

Reply:
I will take as an action to document the above in the `scrapertest` Readme.

Reply:
Sounds good. Thanks for clarifying